user/models.py

```diff
@@ -44,5 +62,12 @@ def __str__(self):
         return str(self.username or 'Unknown')
 
     def save(self, *args, **kwargs):
+        try:
+            User.objects.get(username=self.username)
+            # If the code above is reach on the block try, it means that there
+            # is a duplicate
```

Comment:
BTW, this comment is also a good point as it's not straightfoward to understand. 👍